### ISSUE TITLE
Duotone filter not applied on existing cover block on front end

### ISSUE THREAD

**Issue by kathrynwp** [WordPress/gutenberg#48974]
### Description

The duotone filter is not displayed on the front end on existing cover blocks. The bug is present with GB 15.3.1 but is not present in 15.2.4.

### Step-by-step reproduction instructions

1. On a site running Gutenberg 15.2.4 and WP 6.1.1, add a Cover block to a page
2. Apply a duotone filter to the cover block
3. View the page on the front end

### Result

The duotone filter is visible

4. Now, update Gutenberg to 15.3.1
5. View the page again on the front end

### Result

The duotone effect is no longer visible on the front end, even though it's still visible in the editor

#### Editor view

<img width="1042" alt="Edit_Page_‹_Twenty_Twenty-Three_Testing_—_WordPress" src="https://user-images.githubusercontent.com/7662449/224130655-6d05df9f-323d-436b-9f3d-6f325d907e3f.png">

#### Front end view

<img width="512" alt="Twenty_Twenty-Three_Testing" src="https://user-images.githubusercontent.com/7662449/224130809-6e7cfb46-223b-4fd3-ac5f-6a20d03bf7a1.png">

### Environment info

WordPress 6.1.1
Gutenberg 15.3.1
No other active plugins
Themes: Twenty Twenty-Three, Twenty Twenty-Two, Twenty Twenty-Two

Reported in this forum thread with the Poe theme:

https://wordpress.org/support/topic/update-prevents-duotone-in-frontend/

**Note:** I could _not_ replicate the bug in WP 6.2-RC1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by kathrynwp** [comment]
Also noting that I did not experience the bug on a **new** Cover block, created **after** GB was updated to 15.3.1. Only the pre-existing blocks were affected.

**Comment by getdave** [comment]
cc @ajlende @scruffian

**Comment by bph** [comment]
I can confirm that the duotones on my cover block disappeared after the upgrade to GB 15.3.1.

After the upgrade, the duotone from my old cover block was gone on the frontend. 
After I switched them on and off in the editor, they appeared again on the frontend. 
That seems to be a backwards compatibility bug.  

 @kathrynwp that might be something the OP could try, if it's a finite amount of cover blocks. 

Now in 15.3.1 I also tested to change the duotone highlight and shadow colors, and not use the presets and found they do not get applied on the front end. 

You can see it in the video w/ GB 15.3.1 activated

https://user-images.githubusercontent.com/39980/224145313-e557ebe0-5912-4cf4-91b1-953b2739c848.mp4


If we need to split this in two issues, please let me know and I open a second issue.

**Comment by ajlende** [comment]
@bph/@kathrynwp I'm having a hard time reproducing this issue. Would you be able to share the raw post content that is having issues? You can copy it by switching to the code editor using the three dots options menu in the post editor. That might help me track down the problem.

**Comment by bph** [comment]
@ajlende Here is my cover block content and I uploaded the image as well.. :-) 

to iterate: 

- Have Gutenberg 15.3.1 **deactivated** before you create the cover block. 
- Save the post, 
- Then activate GB 15.3.1
- Then review on the post on the front page first. 

```
<!-- wp:cover {"url":"http://porter-role.local/wp-content/uploads/2023/02/20200809_Taipei_Taiwan_Skyline.jpg","id":15,"dimRatio":50,"isDark":false,"style":{"color":{"duotone":["#345C00","#ffffff"]}}} -->
<div class="wp-block-cover is-light"><span aria-hidden="true" class="wp-block-cover__background has-background-dim"></span><img class="wp-block-cover__image-background wp-image-15" alt="" src="http://porter-role.local/wp-content/uploads/2023/02/20200809_Taipei_Taiwan_Skyline.jpg" data-object-fit="cover"/><div class="wp-block-cover__inner-container"><!-- wp:paragraph {"align":"center","placeholder":"Write title…","textColor":"base","fontSize":"large"} -->
<p class="has-text-align-center has-base-color has-text-color has-large-font-size">testing the cover block</p>
<!-- /wp:paragraph --></div></div>
<!-- /wp:cover -->
```
![20200809_Taipei_Taiwan_Skyline](https://user-images.githubusercontent.com/39980/224185578-5002e14c-8553-4407-9f36-c4652cd2ed67.jpg)

**Comment by druesome** [comment]
Can confirm this on a user's site and my test site as well.

Both sites have Gutenberg version 15.3.1 installed.

**Comment by ajlende** [comment]
I couldn't reproduce the issue because it behaved correctly when `SCRIPT_DEBUG` was `true`, and I had that set in my development environment.

I solved the issue in https://github.com/WordPress/gutenberg/pull/49004. When you are testing, be sure to have `SCRIPT_DEBUG` set to `false` to see the fix.

**Comment by daria2303** [comment]
Another user in the forum reporting the same issue: https://wordpress.com/forums/topic/duotone-filter-visible-in-editor-but-not-in-actual-post/?view=all

**Comment by furqGreen7** [comment]
I seem to have a similar issue, reported [here on the WordPress forum](https://wordpress.com/forums/topic/quadrat-black-theme-broken-colors-wonky/), updated [here](https://wordpress.org/support/topic/update-prevents-duotone-in-frontend/).

**Comment by ajlende** [comment]
@furqGreen7 does the issue exist in trunk for you? There was a regression that was fixed in #50678 and will be in the next Gutenberg release.
